components/core/src/integrations/pluggablefactoryjson.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {JsonMap} from '../util/jsonvalue';
+import {Pluggable} from './pluggable';
+import {PluggableFactory} from './pluggablefactory';
+import {PluggableFactorySet} from './pluggablefactoryset';
+
+export type ModuleNameType = string;
+
+export type PluggableFactoryJsonStanza<
```

Comment:
I added a comment that explains this to the code too. This defines the shape of the JSON config sections where you specify a PluggableFactory. There are 3 different forms it can take: map, array tuple, and just a string if you don't have any params to specify.